indico/modules/events/contributions/models/contributions.py

```diff
@@ -407,3 +410,19 @@ def _add_timetable_consistency_trigger(target, conn, **kw):
         EXECUTE PROCEDURE events.check_timetable_consistency('contribution');
     """.format(target.fullname)
     DDL(sql).execute(conn)
+
+
+@listens_for(Contribution, 'load')
+def _preload_principals(target, context):
+    """
+    Pre-fill contribution with information about ``ContributionPrincipal``s, if available in cache.
+
+    This is set using ``g.contribution_acl_cache`` and is quite useful for queries that involve
+    generating large timetables needing access control information.
+    """
+    if 'contribution_acl_cache' not in g:
```

Comment:
Could also use `cache = g.get('contribution_acl_cache', {})` and avoid the extra check. Doesn't really make a difference though